ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -450,6 +450,20 @@ static void HandleKernelRequest(Message request, void* messageMemory)
                 PrjFS_NotificationType_PreDelete);
             break;
         }
+        
+        case MessageType_KtoU_NotifyFileCreated:
+        {
+            char fullPath[PrjFSMaxPath];
+            CombinePaths(s_virtualizationRootFullPath.c_str(), request.path, fullPath);
+            SetBitInFileFlags(fullPath, FileFlags_IsInVirtualizationRoot, true);
```

Comment:
What if this fails?

Reply:
Thanks for pointing this out, I had not considered that scenario.

I think VFSForGit should log if this happens, what would you think if the kext let the provider know in the NewFileCreated callback (e.g. add a `bool fileSuccessfullyFlagged`)?

As far as impact on VFSForGit, if we fail to set the bit:
- There could be duplicate NewFileCreated notifications
- The kext will not deliver the PreDelete notification (because the file is not flagged)

So far I've not been able to think of a scenario where git would behave incorrectly if this happened, the main issue would be missing (or duplicate callbacks).

Other than logging, do you have any ideas on what else we could\should do if this occurs?

Reply:
This is a general implementation hole that @nickgra is looking into - we need to make sure that the lib's IO is all-or-nothing, and doesn't leave files in a halfway updated state. This one is a bit more problematic, since it's in response to a fileop event and therefore we don't get a chance to redo it. At the very least, we should leave a TODO here, and logging it is a good idea as well. Down the road, we need some strategy for recovering from lost fileop events (and that's a bigger scope than what @nickgra is working on right now, so let's address that one separately)

Reply:
> At the very least, we should leave a TODO here, and logging it is a good idea as well.

I'll leave a TODO for now.  Long term if we can have failures here (once we've stopped using flags) we'll either want a way to let providers know so that they can log the failure to a log file, or we'll want a PrjFSLib log file where we can log it.